free_fleet_adapter/launch/fleet_adapter.launch.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version='1.0' ?>
+
+<launch>
+
+  <arg name="use_sim_time" default="true" description="Use the /clock topic for time to sync with simulation"/>
+  <arg name="config_file" description="The config file that provides important parameters for setting up the adapter"/>
+  <arg name="nav_graph_file" description="The graph that this fleet should use for navigation"/>
+  <arg name="server_uri" default="" description="The URI of the api server to transmit state and task information."/>
+  <arg name="output" default="screen"/>
+
+  <!-- Fleet adapter -->
+  <group>
+    <group if="$(var use_sim_time)">
+      <node pkg="free_fleet_adapter"
+            exec="fleet_adapter.py"
+            args="-c $(var config_file) -n $(var nav_graph_file) -s $(var server_uri) -sim"
```

Comment:
Instead of needing to branch this launch file logic, it may be worth checking if you can tweak its implementation so that you initialize an `rclpy.Node` early on and then check if the `use_sim_time` parameter is set to true or false rather than having a dedicated command line argument for it.

If there's some reason that can't be done, then you could consider tweaking the argument to so that this line could be `--sim $(use_sim_time)` so you can just feed in the boolean instead of needing to have different branches based on whether or not to include `-sim`.

Reply:
That's a good point, I ended up with some conditional variables in the xml instead, https://github.com/open-rmf/free_fleet/pull/145/commits/cccb4033c024b5035a97c7cda146f70970b24baf